src/twister2/plugin.py

```diff
@@ -182,6 +183,21 @@ def pytest_addoption(parser: pytest.Parser):
         action='store_true',
         help='Treat memory overflows as errors.'
     )
+    twister_group.addoption(
+        '-M', '--runtime-artifact-cleanup',
+        choices=('pass', 'all'),
+        help='Cleanup test artifacts. "pass" option only removes artifacts of '
+             'passing or skipping tests. If you wish to remove all artifacts '
+             'including those of failed tests, use "all"'
+    )
+    twister_group.addoption(
+        '--prep-artifacts-for-testing',
```

Comment:
what about extending `--runtime-artifact-cleanup` choices with `device` option?

Reply:
I see that `--prep-artifacts-for-testing` also prepares some files (CMakeCache.txt and runners.yaml) before storing. So it is cleanup + keep binaries + "sanitize paths so files are relocatable"